exporter/exporterhelper/README.md

```diff
@@ -17,12 +17,48 @@ The following configuration options can be modified:
 - `sending_queue`
   - `enabled` (default = true)
   - `num_consumers` (default = 10): Number of consumers that dequeue batches; ignored if `enabled` is `false`
-  - `queue_size` (default = 5000): Maximum number of batches kept in memory before data; ignored if `enabled` is `false`;
+  - `queue_size` (default = 5000): Maximum number of batches kept in memory before data; ignored if `enabled` is `false` or WAL is enabled;
   User should calculate this as `num_seconds * requests_per_second` where:
     - `num_seconds` is the number of seconds to buffer in case of a backend outage
     - `requests_per_second` is the average number of requests per seconds.
+  - `wal_enabled` (default = false): When set, enables Write-Ahead-Log (WAL) via a file extension
 - `resource_to_telemetry_conversion`
   - `enabled` (default = false): If `enabled` is `true`, all the resource attributes will be converted to metric labels by default.
 - `timeout` (default = 5s): Time to wait per individual attempt to send data to a backend.
 
 The full list of settings exposed for this helper exporter are documented [here](factory.go).
+
+
+### WAL
+
+When `wal_enabled` is set, the queue is being buffered to by the storage extension. This has some limitations currently,
+the items that are currently being handled by a consumer are not backed by the persistent storage, which means
+that in case of a sudden shutdown, they might be lost.
+
+```
+                                                   ┌─Consumer #1─┐
+                                                   │    ┌───┐    │
+                         ┌──Deleted──┐        ┌───►│    │ 1 │    ├───► Success
+                         X     x     x        │    │    └───┘    │
+                         x     x     x        │    │             │
+                         x     x     x        │    └─────────────┘
+                         x     x     x        │
+ ┌─────────WAL-backed queue────x─────x───┐    │    ┌─Consumer #2─┐
+ │                       x     x     x   │    │    │    ┌───┐    │
+ │     ┌───┐     ┌───┐ ┌─x─┐ ┌─x─┐ ┌─x─┐ │    │    │    │ 2 │    ├───► Permanent
+ │ n+1 │ n │ ... │ 4 │ │ 3 │ │ 2 │ │ 1 │ ├────┼───►│    └───┘    │      failure
```

Comment:
Maybe add a cross to the arrow next to Permanet Failure to suggest the data does not get out?